Services/ContentService.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Opifer\ContentBundle\Services;
+
+use Opifer\ContentBundle\Model\ContentManager;
+use Doctrine\ORM\EntityManager;
+
+/**
+ * class ContentDuplicator
+ *
+ * @author Vladimir Djuricic
+ */
+class ContentService
+{
+    private $content_manager;
+    private $em;
+
+    /**
+     *
+     * @param ContentManager $content_manager
+     * @param EntityManager $doctrine
+     */
+    public function __construct(ContentManager $content_manager, EntityManager $doctrine)
+    {
+        $this->em = $doctrine;
+        $this->content_manager = $content_manager;
+    }
+    
+    /**
+     * @param Opifer\CmsBundle\Entity\Content $content
+     * @param \Opifer\EavBundle\Entity\Value $nested_in
+     */
+    public function duplicate($content, $nested_in=null)
+    {
+        $contentManager = $this->content_manager;
+
+        //get valueset to clone
+        $valueset = $content->getValueSet();
+
+        //clone valueset
+        $duplicated_valueset = clone $valueset;
+
+        $this->em->detach($duplicated_valueset);
+        $this->em->persist($duplicated_valueset);
+        $this->em->flush();
```

Comment:
You call the following lines three times;

```php
$this->em->detach($duplicated_valueset);
$this->em->persist($duplicated_valueset);
$this->em->flush();
```

I think it would be a bit nicer to move that to a `detach()` method.
Also, is it necessary to call the `persist()` in between? I guess the content item is already persisted here.